crates/ra_hir/src/nameres/collector.rs

```diff
@@ -662,6 +663,14 @@ fn resolve_submodule(
     }
 }
 
+fn resolve_mod_name(name: String) -> String {
```

Comment:
I think we shouldn't do this here. Rather, the fact that `Name` contains an `r#` is a bug: `Name` is supposed to be clean, semantic name. 

I think this cleaning should happen in these two impls: https://github.com/rust-analyzer/rust-analyzer/blob/2b2cd829b0f95aef338227deb05ec7503dae9b6c/crates/ra_hir/src/name.rs#L99-L109